MapboxDirections/Match/MBMatch.swift

```diff
@@ -0,0 +1,74 @@
+    init(matchOptions: MatchingOptions, legs: [RouteLeg], tracepoints: [Tracepoint], distance: CLLocationDistance, expectedTravelTime: TimeInterval, coordinates: [CLLocationCoordinate2D]?, confidence: Float, speechLocale: Locale?) {
+        self.confidence = confidence
+        self.tracepoints = tracepoints
+        super.init(options: matchOptions, legs: legs, distance: distance, expectedTravelTime: expectedTravelTime, coordinates: coordinates, speechLocale: speechLocale)
+    }
+    
+    /**
+     Initializes a new match object with the given JSON dictionary representation and tracepoints.
+     
+     - parameter json: A JSON dictionary representation of the route as returned by the Mapbox Mapbox Map Matching API.
+     - parameter tracepoints: An array of `Tracepoint` that the match found in order.
+     - parameter matchOptions: The `MatchingOptions` used to create the request.
+    */
+    @objc public convenience init(json: [String: Any], tracepoints: [Tracepoint], matchOptions: MatchingOptions) {
+        let legInfo = zip(zip(tracepoints.prefix(upTo: tracepoints.endIndex - 1), tracepoints.suffix(from: 1)),
+                          json["legs"] as? [JSONDictionary] ?? [])
+        let legs = legInfo.map { (endpoints, json) -> RouteLeg in
+            RouteLeg(json: json, source: endpoints.0, destination: endpoints.1, profileIdentifier: matchOptions.profileIdentifier)
+        }
+        
+        let distance = json["distance"] as! Double
+        let expectedTravelTime = json["duration"] as! Double
+        
+        var coordinates: [CLLocationCoordinate2D]?
+        switch json["geometry"] {
+        case let geometry as JSONDictionary:
+            coordinates = CLLocationCoordinate2D.coordinates(geoJSON: geometry)
+        case let geometry as String:
+            coordinates = decodePolyline(geometry, precision: 1e5)!
+        default:
+            coordinates = nil
+        }
+        
+        let confidence = json["confidence"] as! Float
+        
+        var speechLocale: Locale?
+        if let locale = json["voiceLocale"] as? String {
+            speechLocale = Locale(identifier: locale)
+        }
+        
+        self.init(matchOptions: matchOptions, legs: legs, tracepoints: tracepoints, distance: distance, expectedTravelTime: expectedTravelTime, coordinates: coordinates, confidence: confidence, speechLocale: speechLocale)
+    }
+    
+    /**
+     A number between 0 and 1 that indicates the Map Matching API’s confidence that the match is accurate. A higher confidence means the match is more likely to be accurate.
+     */
+    @objc open var confidence: Float
+    
+    
+    @objc let tracepoints: [Tracepoint]?
+    
+    public var matchOptions: MatchingOptions {
+        return super.directionsOptions as! MatchingOptions
+    }
+    
+    @objc public required convenience init?(coder decoder: NSCoder) {
+        self.init(coder: decoder)
+        confidence = decoder.decodeFloat(forKey: "confidence")
+    }
```

Comment:
Also encode and decode the tracepoints, and potentially override the equality methods to consider whether the tracepoints are equal.